qiskit_ibm_runtime/base_runtime_job.py

```diff
@@ -220,7 +223,7 @@ def error_message(self) -> Optional[str]:
 
     def _set_status_and_error_message(self) -> None:
         """Fetch and set status and error message."""
-        if not self.in_final_state():
+        if self._status not in self.JOB_FINAL_STATES:
```

Comment:
not sure why this change is needed, although I assume there is a good reason.

Reply:
It has to be done this way because `_set_stauts_and_error_message` is called in `status` so calling `in_final_state` would result in an infinite loop